lib/chef-dk/skeletons/code_generator/recipes/build_cookbook.rb

```diff
@@ -120,47 +120,41 @@
   execute('git-create-feature-branch') do
     command('git checkout -t -b add-delivery-configuration')
     cwd delivery_project_dir
-
-    not_if 'git branch |grep "add-delivery-configuration"'
+    not_if { shell_out('git branch', cwd: delivery_project_dir).stdout.match(/add-delivery-configuration/) }
   end
 
   execute('git-add-delivery-config-json') do
     command('git add .delivery/config.json')
     cwd delivery_project_dir
-
-    only_if 'git status -u --porcelain |grep ".delivery/config.json"'
+    only_if { shell_out('git status -u --porcelain', cwd: delivery_project_dir).stdout.match(%r{.delivery/config.json}) }
   end
 
   # Adding the new prototype file to the feature branch
   # so it gets checked in with the delivery config commit
   execute('git-add-delivery-project-toml') do
     command('git add .delivery/project.toml')
     cwd delivery_project_dir
-
-    only_if 'git status -u --porcelain |grep ".delivery/project.toml"'
+    only_if { shell_out('git status -u --porcelain', cwd: delivery_project_dir).stdout.match(%r{.delivery/project.toml}) }
   end
 
   execute('git-commit-delivery-config') do
     command('git commit -m "Add generated delivery configuration"')
     cwd delivery_project_dir
-
-    only_if 'git status -u --porcelain | egrep "config.json|project.toml"'
+    only_if { shell_out('git status -u --porcelain', cwd: delivery_project_dir).stdout.match(/config\.json|project\.toml/) }
```

Comment:
Does the pipe works the same way than in `egrep`? I mean, is it searching for either `config.json` or `project.toml`? 🤔